### DISCUSSION TITLE
Проверка на пустоту при отправке файлов

### DISCUSSION THREAD

**Discussion by k1erk** [totumonline/totum-ru-issues-and-discussions#178] [Как это сделать?] [answered]
День добрый, подскажите пожалуйста решение. Суть в следующем:
С помощью временных таблиц был организован обмен документов. Новому сотруднику отправляется ссылка на форму где он прикрепляет файлы в формате pdf для приема на работу и далее эти документы отправляются через emalSend в кадры.
Так вот есть документы которые  могут быть, а могут и не быть. Военный билет не у каждой девушки есть. 

Формирование вложения и генерации названиям файлом происходит через 
```
rowCreate(field: "voennik.pdf" = #h_voennyy_bilet[0][file]; field: "deti.pdf" = #h_svidetelstvo_o_rojdenii_detey[0][file])
```

Таких критериев 4:

+ дети - дети есть присылаем свид. рождения нет значит не генерируем файл при отправке
+ карта - аналогично детям
+ военник - аналогично детям
+ фамилия - фамилия не менялась тоже не генерируем свид. о браке

Какой может быть изящный метод не генерировать пустые файлы если нет прикрепленных файлов и стоит триггер что документ отсутствует. Спасибо за отклик.

![image](https://user-images.githubusercontent.com/110602346/182825208-9524ad18-3909-4b06-91ad-ef191e9660db.png)

**Comment by ttm-alexeypolunin** [comment, marked as answer]
Вот такой есть подход к набору `row` по условиям:

```
= : rowAdd(row: $base_row; row: $if_1; row: $if_2)

	base_row: rowCreate(field: "data_1" = "data_1"; field: "data_2" = "data_2")
	
	if_1: if(condition: #field_if_1 != ""; then: $row_if_1; else: $#lc)
		row_if_1: rowCreate(field: "data_3" = "data_3")
	if_2: if(condition: #field_if_2 != ""; then: $row_if_2; else: $#lc)
		row_if_2: rowCreate(field: "data_4" = "data_4")
```

`base_row` — то, что у вас будет всегда.

К нему вы добавляете то, что по условиям через `rowAdd`.

В `if` вы передаете либо `row` с тем значением, которое вам нужно добавить или переписать или пустой список `$#lc`.

В вашем случае, если проверка в if будет идти с полем Файл, то надо сравнивать с пустым списком `#field_if_1 != $#lc` тк в поле Файл хранится список и вам нужно проверить пустой ли он или нет. Если сравнится с простой пустотой то есть случаи, когда это сравнение пройдет, даже если там будет файл. Если контроль будет идти по какому-то триггеру типа чекбокса, то тогда просто с `true`/`false`.

**Reply by ttm-alexeypolunin** [reply]
Еще придумал, конкретно для этого случая простую штуку — если поля пустые, то и значения в row будут пустые. Надо фильтрануть row по пустым значениям:

```
= : listFilter(list: $base_row; key: "value" != "")
	
	base_row: rowCreate(field: "data_1" = "data_1"; field: "data_2" = ""; field: "data_3" = "data_3"; field: "data_4" = "")

```

**Reply by ttm-alexeypolunin** [reply]
И в догонку — можно и поля визуально скрыть по условиям из таблицы, используя параметр `fieldhide` функции `setTableFormat` в форматировании таблицы.

**Reply by k1erk** [reply]
Добрый день. Все получилось и работает. Удачи вам в развитии. Спасибо.
